server/authentication_test.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Marc Berhault (marc@cockroachlabs.com)
+
+package server
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/kv"
+	"github.com/cockroachdb/cockroach/testutils"
+	"github.com/cockroachdb/cockroach/ts"
+)
+
+// Verify client certificate enforcement.
+func TestSSLEnforcement(t *testing.T) {
+	s := StartTestServer(t)
+	defer s.Stop()
+	testCases := []struct {
+		method, key   string
+		certsStatus   int // Status code for https with client certs.
+		noCertsStatus int // Status code for https without client certs.
+	}{
+		// /ui/: basic file server: no auth.
+		{"GET", "/index.html", http.StatusOK, http.StatusOK},
```

Comment:
I did earlier, until I realized that I didn't want to be testing the internal handlers at all, just the ssl verification wrapper.
mainly, if you want to return StatusUnauthorized for specific methods, I should be fine with it.